src/core/public/workspace/workspaces_service.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { BehaviorSubject, combineLatest } from 'rxjs';
+import { isEqual } from 'lodash';
+
+import { CoreService, WorkspaceAttribute } from '../../types';
+
+type WorkspaceObject = WorkspaceAttribute & { readonly?: boolean };
+
+export interface WorkspaceObservables {
+  currentWorkspaceId$: BehaviorSubject<string>;
+  currentWorkspace$: BehaviorSubject<WorkspaceObject | null>;
+  workspaceList$: BehaviorSubject<WorkspaceObject[]>;
+  workspaceEnabled$: BehaviorSubject<boolean>;
```

Comment:
I wonder if this is redundant? Since a plugin being disabled is kind of a before run time operation.

When setting this up as well, the workspaces plugin can set the value for `/api/core/capabilities` and  plugins can check it is available to use
http://localhost:5609/vzu/api/core/capabilities

Reply:
That's a very good point. I will update to use capabilities

Reply:
@kavilla Just to confirm that you mean we could use `core.capabilities.registerProvider` to register a flag when workspace plugin is loaded. Then we could read this at frontend via, for example `application.capabilities.workspaces.loaded`

Reply:
Cool. I've fine with being part of a backlog but to be revisited prior to 2.11

cc: @manasvinibs  (since she has worked a lot with it)